### ISSUE TITLE
Some collective ops seem to dislike gradient checkpoints in some cases

### ISSUE THREAD

**Issue by justheuristic** [BlackSamorez/tensor_parallel#29]
In the T5 tutorial, gradient checkpointing works fine (and correctly) in a custom training loop, but running it inside Trainer causes "trying to backward a second time" error.
